theano/tensor/signal/pool.py

```diff
@@ -443,13 +450,15 @@ def c_code(self, node, name, inp, out, sub):
           %(z)s = (PyArrayObject*) PyArray_ZEROS(4, dims, typenum,0);
         }
         // used for indexing a pool region inside the input
-        int r_st, r_end, c_st, c_end;
         dtype_%(x)s collector; // temp var for the value in a region
         if (z_r && z_c)
         {
-            for(int b=0; b<PyArray_DIMS(%(x)s)[0]; b++){
-              for(int k=0; k<PyArray_DIMS(%(x)s)[1]; k++){
-                for(int i=0; i< z_r; i++){
+            int r_st, r_end, c_st, c_end;
+            %(omp_parallel)s
+            for(int t = 0; t < PyArray_DIMS(%(x)s)[0] * PyArray_DIMS(%(x)s)[1]; t++){
+                int b = t / PyArray_DIMS(%(x)s)[1];
+                int k = t %% PyArray_DIMS(%(x)s)[1];
```

Comment:
I would swap the / and %%. Currently, if a thread need to do the iteration i and i+1, it will read memory that is less contiguous in memory then if we swap them. As the schedule is static, this should happen frequently. So I think it would be more cache friendly to swap the b and k here.